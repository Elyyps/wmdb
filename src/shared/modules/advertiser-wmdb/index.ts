export * from './advertiser-wmdb.component'