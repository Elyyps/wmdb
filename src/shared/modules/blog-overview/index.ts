export * from "./blog-overview.component";
