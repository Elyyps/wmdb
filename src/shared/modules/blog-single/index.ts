export * from "./blog-single.component";
