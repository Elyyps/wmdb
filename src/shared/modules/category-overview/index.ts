export * from "./category-overview.component";
