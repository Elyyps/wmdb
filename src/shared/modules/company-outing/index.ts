export * from "./company-outing.component";
