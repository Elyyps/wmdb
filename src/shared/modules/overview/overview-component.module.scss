.overview {
  padding: 40px 0 72px 0;
  @media (max-width: $breakpoint-xsmall-max) {
    padding: 16px 0;
  }
  &-head {
    &__subtitle {
      display: flex;
      width: 100%;
      justify-content: space-between;
      a {
        text-decoration: none;
      }
      span {
        font-size: 14px;
        line-height: 1.41;
        color: #6a6a6a;
      }
    }
    &__list {
      display: flex;
      align-items: center;
      margin: 10px -2px -2px -2px;
      flex-wrap: wrap;
      > li {
        padding: 0 2px 2px 2px;
      }
    }
  }
  &-head {
    margin-bottom: 24px;
    @media (max-width: $breakpoint-xsmall-max) {
      margin-bottom: 16px;
    }
  }
  &-body {
    .uk-list-check {
      margin: 0;
      padding: 24px 0;
      display: flex;
      align-items: center;
      justify-content: center;
      border-top: 1px solid $gray-light-secondary;
      border-bottom: 1px solid $gray-light-secondary;
      @media (max-width: $breakpoint-xsmall-max) {
        background-color: #fff8ed;
        margin: 0 -16px;
        padding: 16px;
        margin-bottom: 16px;
      }
      > li {
        margin: 0 20px;
        @media (max-width: $breakpoint-medium-max) {
          margin: 0 5px;
        }
        @media (max-width: $breakpoint-xsmall-max) {
          margin: 0;
          text-align: center;
          justify-content: center;
          &:not(:first-child) {
            display: none;
          }
        }
        &:not(:last-child) {
          margin-bottom: 0;
        }
      }
    }
    .card {
      &-post {
        @media (max-width: $breakpoint-xsmall-max) {
          padding: 0 0 24px 0;
        }
      }
      &-advertise {
        @media (max-width: $breakpoint-xsmall-max) {
          margin: 0 0 24px 0;
        }
      }
    }
  }
  &-grid {
    display: flex;
    flex-wrap: wrap;
    margin: 0 -16px;
    > * {
      padding: 0 16px;
      box-sizing: border-box;
      &:first-child {
        flex-basis: 296px;
      }
    }
  }
}
