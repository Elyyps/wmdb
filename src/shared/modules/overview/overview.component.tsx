import * as React from "react";
import { OverviewFilterComponent } from "@app/core/overview-filter";
import { LabelComponent } from "./label/label.component";
import { CardContainer } from "@app/core/card-post";
import { PaginationComponent } from "@app/core/pagination";
import { ListCheckComponent } from "@app/core/list-check";
import { IOverviewFilterItem, IOutingCard, getCardsPaginated } from "@app/api/modules/overview";
import { GenerateDummyFilterOverview } from "../../api/modules/overview/dummy-data";

export interface IOverviewComponentProps {}

const TAKE = 8;
const AD_POSITION = 6;

const OverviewComponent = () => {
  const rangeMax = 200;
  const [currentPage, setCurrentPage] = React.useState(1);
  const [totalPages, setTotalPages] = React.useState(0);
  const [cards, setCards] = React.useState<IOutingCard[]>([]);
  const [currentFilter, setCurrentFilter] = React.useState<IOverviewFilterItem>({
    keyword: "",
    checkedItems: [],
    filterText: "",
    range: 0
  });

  const nextPage = () => {
    if (currentPage < totalPages) setCurrentPage(currentPage + 1);
  };
  const previousPage = () => {
    if (currentPage > 1) setCurrentPage(currentPage - 1);
  };

  const handleClose = (item: any) => {
    const newCheckedItems = [...currentFilter.checkedItems];
    newCheckedItems.splice(item, 1);
    setCurrentFilter({ ...currentFilter, checkedItems: newCheckedItems });
  };
  const handleCloseTextFilter = () => {
    const newCheckedItems = { ...currentFilter };
    setCurrentFilter({ ...newCheckedItems, filterText: "" });
  };
  const handleCloseKeyword = () => {
    const newCheckedItems = { ...currentFilter };
    setCurrentFilter({ ...newCheckedItems, keyword: "" });
  };
  const handleCloseRange = () => {
    const newCheckedItems = { ...currentFilter };
    setCurrentFilter({ ...newCheckedItems, range: 0 });
  };

  React.useEffect(() => {
    const cardsFiltered = getCardsPaginated(0, TAKE, {
      keyword: "",
      filterText: "",
      checkedItems: [],
      range: rangeMax
    });
    setTotalPages(cardsFiltered.total);
    setCards(cardsFiltered.cards);
  }, []);
  React.useEffect(() => {
    const skip = (currentPage - 1) * TAKE;
    const cardsFiltered = getCardsPaginated(skip, TAKE, currentFilter);
    setTotalPages(cardsFiltered.total);
    setCards(cardsFiltered.cards);
  }, [currentPage]);

  React.useEffect(() => {
    const skip = (currentPage - 1) * TAKE;
    const cardsFiltered = getCardsPaginated(skip, TAKE, currentFilter);
    setTotalPages(cardsFiltered.total);
    setCards(cardsFiltered.cards);
  }, [currentFilter]);

  return (
    <div className="overview">
      <div className="uk-container">
        <div className="uk-grid">
          <div className="uk-width-1-4@m">
            <OverviewFilterComponent
              searchPlaceholder="Plaats, regio of provincie"
              range={0}
              sidebarList={{ title: "Handige links", list: ["Link 1", "Link 2"] }}
              currentFilter={currentFilter}
              onFilterChange={changed => {
                setCurrentFilter(changed);
              }}
              filterItems={GenerateDummyFilterOverview()}
              rangeMax={rangeMax}
            />
          </div>
          <div className="uk-width-3-4@m">
            <div className="overview-head uk-visible@m">
              {currentFilter.checkedItems.length > 0 && (
                <div className="overview-head__subtitle">
                  {`${currentFilter.checkedItems.length +
                    (currentFilter.filterText ? 1 : 0) +
                    (currentFilter.range ? 1 : 0)}  uitjes gevonden gebaseerd op de volgende filters:`}
                </div>
              )}

              <ul className="overview-head__list" data-uk-margin>
                {currentFilter.filterText && (
                  <li>
                    <LabelComponent
                      onClick={handleCloseTextFilter}
                      label={`Plaats, regio of provincie: ${currentFilter.filterText}`}
                    />
                  </li>
                )}
                {currentFilter.keyword && (
                  <li>
                    <LabelComponent onClick={handleCloseKeyword} label={`Keyword: ${currentFilter.keyword}`} />
                  </li>
                )}
                {currentFilter.range > 0 && (
                  <li>
                    <LabelComponent onClick={handleCloseRange} label={`${currentFilter.range} personen`} />
                  </li>
                )}
                {currentFilter.checkedItems &&
                  Object.keys(currentFilter.checkedItems).map((item: any, key: number) => (
                    <li key={key}>
                      <LabelComponent
                        onClick={() => handleClose(item)}
                        label={currentFilter.checkedItems[item].label}
                      />
                    </li>
                  ))}
              </ul>
            </div>

            {totalPages ? (
              <div className="overview-body">
                <ListCheckComponent
                  labels={["Kwalitatief aanbod van uitjes", "Direct contact", "Unieke content van uitjes"]}
                />
                <CardContainer Cards={cards.slice(0, AD_POSITION)} />
                {cards.length >= AD_POSITION && (
                  <>
                    <ListCheckComponent
                      labels={["Kwalitatief aanbod van uitjes", "Direct contact", "Unieke content van uitjes"]}
                    />
                    <CardContainer Cards={cards.slice(AD_POSITION, cards.length)} />
                  </>
                )}
                {totalPages > 1 && (
                  <PaginationComponent
                    changePage={setCurrentPage}
                    previousPage={previousPage}
                    nextPage={nextPage}
                    currentPage={currentPage}
                    totalPages={totalPages}
                  />
                )}
              </div>
            ) : (
              "Nothing to show"
            )}
          </div>
        </div>
      </div>
    </div>
  );
};
export { OverviewComponent };
