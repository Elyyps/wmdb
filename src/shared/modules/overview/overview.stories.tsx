
import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { OverviewComponent } from "./overview.component";
import { generateDummyOverviewModule } from "@app/api/modules/overview";

storiesOf("Overview", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <OverviewComponent overviewModule={generateDummyOverviewModule()} />);

  