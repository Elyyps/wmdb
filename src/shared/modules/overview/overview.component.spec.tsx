import React from "react";
import { shallow } from "enzyme";
import { OverviewComponent } from "./overview.component";
import { generateDummyOverviewModule } from "@app/api/modules/overview";

describe("[Overview]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<OverviewComponent overviewModule={generateDummyOverviewModule()} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
