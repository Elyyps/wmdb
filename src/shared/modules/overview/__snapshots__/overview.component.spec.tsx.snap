// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Overview] should render component without crashing 1`] = `
<div
  className="overview"
>
  <div
    className="uk-container"
  >
    <div
      className="uk-grid"
    >
      <div
        className="uk-width-1-4@m"
      >
        <OverviewFilterComponent
          currentFilter={
            Object {
              "checkedItems": Array [],
              "filterText": "",
              "keyword": "",
              "range": 0,
            }
          }
          filterItems={
            Array [
              Object {
                "checkboxes": Array [
                  Object {
                    "count": 30,
                    "id": 1,
                    "label": "Citygames",
                    "name": "citygames",
                  },
                  Object {
                    "count": 30,
                    "id": 2,
                    "label": "Beachgames",
                    "name": "Beachgames",
                  },
                  Object {
                    "count": 30,
                    "id": 3,
                    "label": "Indoor Spel en Ontspanning",
                    "name": "IndoorSpelenOntspanning",
                  },
                  Object {
                    "count": 30,
                    "id": 4,
                    "label": "Outdoor Spel en Ontspanning",
                    "name": "Outdoor Spel en Ontspanning",
                  },
                ],
                "title": "Spel & Ontspanning",
              },
              Object {
                "checkboxes": Array [
                  Object {
                    "count": 30,
                    "id": 5,
                    "label": "Fun op Wielen",
                    "name": "Funopwielen",
                  },
                  Object {
                    "count": 30,
                    "id": 6,
                    "label": "Outdoor Lucht",
                    "name": "Outdoor Lucht",
                  },
                  Object {
                    "count": 30,
                    "id": 7,
                    "label": "Outdoor Water",
                    "name": "outdoorwater",
                  },
                  Object {
                    "count": 30,
                    "id": 8,
                    "label": "Indoor Sport en Actie",
                    "name": "indorsportenactie",
                  },
                  Object {
                    "id": 26,
                    "label": "Outdoor Sport en Actie",
                    "name": "outdorsportenactie",
                  },
                ],
                "title": "Sportief & Actie(f)",
              },
              Object {
                "checkboxes": Array [
                  Object {
                    "count": 30,
                    "id": 9,
                    "label": "Creatief",
                    "name": "creatief",
                  },
                  Object {
                    "count": 30,
                    "id": 10,
                    "label": "Theater",
                    "name": "theater",
                  },
                  Object {
                    "count": 30,
                    "id": 11,
                    "label": "Muziek en Dans",
                    "name": "muziekenDans",
                  },
                ],
                "title": "Creatieve Uitjes",
              },
              Object {
                "checkboxes": Array [
                  Object {
                    "count": 30,
                    "id": 12,
                    "label": "Kookworkshops",
                    "name": "kookworkshops",
                  },
                  Object {
                    "count": 30,
                    "id": 13,
                    "label": "Kunst en Cultuur",
                    "name": "kunstencultuur",
                  },
                  Object {
                    "count": 30,
                    "id": 14,
                    "label": "Body en Spirit",
                    "name": "bodyenspirit",
                  },
                ],
                "title": "Workshops",
              },
              Object {
                "checkboxes": Array [
                  Object {
                    "count": 30,
                    "id": 15,
                    "label": "Eten en drinken",
                    "name": "etenendrinken",
                  },
                  Object {
                    "count": 30,
                    "id": 16,
                    "label": "Themafeesten",
                    "name": "Themafeesten",
                  },
                  Object {
                    "count": 30,
                    "id": 17,
                    "label": "Totaalprogramma",
                    "name": "totaalprogramma",
                  },
                  Object {
                    "count": 30,
                    "id": 18,
                    "label": "Attracties en Entertainment",
                    "name": "Attractiesenentertainment",
                  },
                ],
                "title": "Eten, Drinken & Feesten",
              },
              Object {
                "checkboxes": Array [
                  Object {
                    "count": 30,
                    "id": 19,
                    "label": "Teamtraining",
                    "name": "Teamtraining",
                  },
                  Object {
                    "count": 30,
                    "id": 20,
                    "label": "Teambuilding",
                    "name": "Teambuilding",
                  },
                  Object {
                    "count": 30,
                    "id": 21,
                    "label": "Uit Vergaderen",
                    "name": "Uit Vergaderen",
                  },
                ],
                "title": "Teambuilden",
              },
              Object {
                "checkboxes": Array [
                  Object {
                    "count": 30,
                    "id": 22,
                    "label": "Wintersport",
                    "name": "wintersport",
                  },
                  Object {
                    "count": 30,
                    "id": 23,
                    "label": "Stedentrips",
                    "name": "stedentrips",
                  },
                  Object {
                    "count": 30,
                    "id": 24,
                    "label": "Adventure & Outdoor",
                    "name": "adventureeoutdoor",
                  },
                  Object {
                    "count": 30,
                    "id": 25,
                    "label": "Reizen Buitenland",
                    "name": "reizenbuitenland",
                  },
                ],
                "title": "Incentives",
              },
            ]
          }
          onFilterChange={[Function]}
          range={0}
          rangeMax={200}
          searchPlaceholder="Plaats, regio of provincie"
          sidebarList={
            Object {
              "list": Array [
                "Link 1",
                "Link 2",
              ],
              "title": "Handige links",
            }
          }
        />
      </div>
      <div
        className="uk-width-3-4@m"
      >
        <div
          className="overview-head uk-visible@m"
        >
          <ul
            className="overview-head__list"
            data-uk-margin={true}
          />
        </div>
        Nothing to show
      </div>
    </div>
  </div>
</div>
`;
